src/source-control/GitSourceControl.ts

```diff
@@ -38,49 +53,396 @@ export class GitSourceControl implements SourceControl {
   }
 
   loadRepositoryInformation(): void {
-    // TODO: Implement
+    this.repositoryUpdateListener(this.repo);
+  }
+
+  private async loadIfChangesPresent() {
+    // If commitHashMap is populated or changes are not detected, we need to return
+    if (!this.repo) {
+      return;
+    }
+    const repo = await nodegit.Repository.open(this.repoPath);
+    const repoStatus = await repo.getStatus();
+    if (repoStatus.length === 0) {
+      // No changes in repo, can return
+      return;
+    } else {
+      await this.populateGitSourceControl();
+      this.loadRepositoryInformation();
+    }
+    // Else we can call populate GitSourceControl
+  }
+
+  private async populateGitSourceControl(): Promise<void> {
+    const repo = await nodegit.Repository.open(this.repoPath);
+    const repoStatus = await repo.getStatus();
+    const refs = await repo.getReferences();
+    const headHash = (await repo.getHeadCommit()).sha();
+    const hashMap = new Map<CommitHash, Commit>();
```

Comment:
This is a really generic name 😆 Could we rename it to something like `commitHashMap` or `mapOfCommitHashesToCommits` or something similar? The name "hash" is pretty unfortunate since it has 2 meanings in our code (hashmap and commit hash)

Reply:
Named it commitHashMap 👍